dataphos-persistor/values.yaml

```diff
@@ -1,10 +1,185 @@
+## @param images.indexer The Indexer image.
+## @param images.indexerApi The Indexer API image.
+## @param images.indexerMongoDb The Mongo image to be used by the indexer.
+## @param images.resubmitter The Resubmitter image.
+##
 images:
   persistor: syntioinc/dataphos-persistor-core:1.0.0
   indexer: syntioinc/dataphos-persistor-indexer:1.0.0
   indexerApi: syntioinc/dataphos-persistor-indexer-api:1.0.0
   indexerMongoDb: mongo:4.0
   resubmitter: syntioinc/dataphos-persistor-resubmitter:1.0.0
+
+## The object containing the information on all of the indexer(s) to be deployed as part of the Helm installation.
+##
+indexer:
+
+  ## The object representing the individual indexer's configuration.
+  ## @param IDX_ID.broker Reference to the broker the indexer metadata is retrieved from.
+  ## @param IDX_ID.topic The topic the indexer metadata are pulled from.
+  ## @param IDX_ID.consumerID The consumer identifier (subscription, consumer group, etc.)
+  ## @param IDX_ID.deadletterTopic The name of the dead-letter topic used for failed data processing. If not set, dead-lettering is disabled (must be enabled if indexer.IDX_ID.broker == kafka).
+  ## @param IDX_ID.replicas The number of replicas of the indexer API microservice.
+  ## @param IDX_ID.consumerReplicas The number of replicas of a given indexer instance to pull/process messages simultaneously.
+  ## @param IDX_ID.storageSize The size of the Indexer's underlying Mongo storage.
+  ## @param IDX_ID.batchSize Maximum number of messages in a batch.
+  ## @param IDX_ID.batchMemory Maximum bytes in batch.
+  ## @param IDX_ID.batchTimeout Time to wait before writing batch if it is not full.
+  ##
+  kafkaIndexer:
+    broker: someKafkaBroker
+    topic: idx-topic
+    consumerID: idx-topic-sub
+    deadletterTopic: idx-dl-topic
+    replicas: 1 # Scale the service exposing the metadata
+    consumerReplicas: 1 # Scale the consumers reading and storing the metadata
+    storageSize: 1Gi
+    batchSize: 5000
+    batchMemory: 1000000
+    batchTimeout: 30s
+  ## @param IDX_ID.serviceAccountSecret The reference to a secret that contains a key.json key and the contents of a Google Service Account JSON file as its contents.
+  ##
+  pubsubIndexer:
+    broker: somePubSubBroker
+    topic: idx-topic
+    consumerID: idx-topic-sub
+    deadletterTopic: idx-dl-topic
+    replicas: 1 # Scale the service exposing the metadata
+    consumerReplicas: 1 # Scale the consumers reading and storing the metadata
+    storageSize: 1Gi
+    serviceAccountSecret:
+  ## @param IDX_ID.mongoConnectionString An optional MongoDB connection string if using an external database. If this value is not set, a MongoDB instance will be deployed on the same cluster.
+  ##
+  externalMongoIndexer:
+    broker: somePubSubBroker
+    topic: idx-topic
+    consumerID: idx-topic-sub
+    deadletterTopic: idx-dl-topic
+    replicas: 1
+    consumerReplicas: 1
+    storageSize: 1Gi
+    mongoConnectionString:
+
```

Comment:
Put this into comments and leave a meaningful example (e.g. leave only Kafka examples with Persistor using ABS while leaving all other examples in the comments)
I think we should add an additional comment saying "If you want a new instance of Indexer, comment out this section", or something along those lines.
The same comment goes for other similar situations where we have objects that create instances (e.g. gcsPersistor, absResubmitter, etc.).